_episodes_rmd/01-raster-structure.Rmd

```diff
@@ -1,406 +1,498 @@
+  ggtitle("Orthographic Imagery", subtitle = "Blue band, with NA highlighted") +
+  coord_equal()
+```
+
+The value that is conventionally used to take note of missing data (the
+`NoDataValue` value) varies by the raster data type. For floating-point rasters,
+the figure `-3.4e+38` is a common default, and for integers, `-9999` is also
+frequently seen. Some disciplines have specific conventions that vary from these
+common values.
+
+In some cases, other `NA` values may be more appropriate. An `NA` value should
+be a) outside the range of valid values, and b) a value that fits the data type
+in use. For instance, if your data ranges continuously from -20 to 100, `0` is
+not an acceptable `NA` value! Or, for categories that number 1-15, `0` might be
+fine for `NA`, but using `-.000003` will force you to save the GeoTIFF on disk
+as a floating point raster, resulting in a bigger file.
+
+If we are lucky, our GeoTIFF file has a tag that tells us what is the
+`NoDataValue`. If we are less lucky, we can find that information in the
+raster's metadata. If a `NoDataValue` was stored in the GeoTIFF tag, when `R`
+opens up the raster, it will assign each instance of the value to `NA`. Values
+of `NA` will be ignored by `R` as demonstrated above.
+
+> ## Challenge
+> Use the output from the `GDALinfo()` function to find out what `NoDataValue` is used for our DSM_HARV dataset.
+>
+> > ## Answers
+> >
+> > ```{r}
+> > GDALinfo("data/NEON-DS-Airborne-Remote-Sensing/HARV/DSM/HARV_dsmCrop.tif")
+> > ```
+> {: .solution}
+{: .callout}
+
+## Bad Data Values in Rasters
+
+Bad data values are different from `NoDataValue`s. Bad data values are values
+that fall outside of the applicable range of a dataset.
+
+Examples of Bad Data Values:
+
+* The normalized difference vegetation index (NDVI), which is a measure of
+greenness, has a valid range of -1 to 1. Any value outside of that range would
+be considered a "bad" or miscalculated value.
+* Reflectance data in an image will often range from 0-1 or 0-10,000 depending
+upon how the data are scaled. Thus a value greater than 1 or greater than 10,000
+is likely caused by an error in either data collection or processing.
+
+### Find Bad Data Values
+Sometimes a raster's metadata will tell us the range of expected values for a
+raster. Values outside of this range are suspect and we need to consider than
+when we analyze the data. Sometimes, we need to use some common sense and
+scientific insight as we examine the data - just as we would for field data to
+identify questionable values.
+
+Plotting data with appropriate highlighting can help reveal patterns in bad
+values and may suggest a solution. Below, reclassification is used to highlight
+elevation values over 400m with a contrasting colour.
+
+```{r 'demo-bad-data-highlighting'}
+# reclassify raster to ok/not ok
+DSM_highvals <- reclassify(DSM_HARV, rcl = c(0, 400, NA_integer_, 400, 420, 1L), include.lowest = TRUE)

```

Comment:
This line is pretty complex. Learners will need to have it broken down for them.